addons/web/controllers/main.py

```diff
@@ -702,8 +702,9 @@ def backup(self, master_pwd, name, backup_format = 'zip'):
     @http.route('/web/database/restore', type='http', auth="none", methods=['POST'], csrf=False)
     def restore(self, master_pwd, backup_file, name, copy=False):
         try:
-            data = base64.b64encode(backup_file.read())
-            dispatch_rpc('db', 'restore', [master_pwd, name, data, str2bool(copy)])
+            data = ''
+            for chunk in iter(lambda: backup_file.read(8190), b''):
+                data += base64.b64encode(chunk)            dispatch_rpc('db', 'restore', [master_pwd, name, data, str2bool(copy)])
```

Comment:
Oops, my mistake.

Could you elaborate on data being a bytearray? As I see it, `dispatch_rpc()` needs `data` to be a base64 encoded string. Or maybe I'm missing something.